### PR TITLE
chore: update third-party actions to address deprecations

### PR DESCRIPTION
**Pull request by BRKalow**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

Updates third-party actions to address deprecation warnings.

### CODE REVIEW

**Comment by BRKalow** [comment]
That doesn't look right ☝️ 😄

File: .github/workflows/docker_publish.yml [hunk 1]
```diff
@@ -19,18 +19,17 @@ jobs:
...
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       # Emulate ARM Processor
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
-      # Workaround: https://github.com/docker/build-push-action/issues/461
+        uses: docker/setup-qemu-action@v2
       - name: Setup Docker buildx
-        uses: docker/setup-buildx-action@79abd3f86f79a9d68a23c75a09a9a85889262adf
```
Comment (kendallstrautman) [inline .github/workflows/docker_publish.yml:28]: 🤔 was the tagging for the specific hash related to that issue commented above?

**Review by kendallstrautman — APPROVED** [review]
thank you!

File: .github/workflows/docker_publish.yml (hunk 1 above)
Comment (BRKalow) [reply .github/workflows/docker_publish.yml:28]: I _think_ so. I had pulled most of this from a template, but I figured I'd give the latest versions a try as the issue is closed.

**Pull request merged**